### PR TITLE
Push for matrixprofile-ts v 0.0.7

### PR DESCRIPTION
**Pull request by vanbenschoten**

### CODE REVIEW

**Comment by vanbenschoten** [comment]
@tylerwmarrs per my Discord message, would you mind serving as a second set of eyes on this?

File: matrixprofile/matrixProfile.py [hunk 1]
```diff
@@ -17,11 +17,62 @@
...
+    Helper function to determine if a value is array like.
+    Parameters
+    ----------
+    a : obj
+        Object to test.
+    Returns
+    -------
+    True or false respectively.
+    """
+    return isinstance(a, (list, tuple, np.array))
```
Comment (tylerwmarrs) [inline matrixprofile/matrixProfile.py]: np.array SHOULD BE np.ndarray

`return isinstance(a, tuple([list, tuple, np.ndarray]))`

File: matrixprofile/matrixProfile.py [hunk 2]
```diff
@@ -52,6 +103,9 @@ def _matrixProfile(tsA,m,orderClass,distanceProfileFunction,tsB=None):
     order = orderClass(len(tsA)-m+1)
     mp, mpIndex = _self_join_or_not_preprocess(tsA, tsB, m)
 
+    tsA = _clean_nan_inf(tsA)
```
Comment (tylerwmarrs) [inline matrixprofile/matrixProfile.py]: All of these checks will fail in the case of tsB being None. You need to check for a self join first and copy tsA to tsB. After that you can call _clean_nan_inf(tsA) etc.

```python
if not is_array_like(tsB):
    tsB = tsA
```

File: matrixprofile/matrixProfile.py [hunk 3]
```diff
@@ -83,26 +137,29 @@ def _stamp_parallel(tsA, m, tsB=None, sampling=0.2, n_threads=-1, random_state=N
     """
     if n_threads is -1:
         n_threads = multiprocessing.cpu_count()
-    
+
     n = len(tsA)
     mp, mpIndex = _self_join_or_not_preprocess(tsA, tsB, m)
```
Comment (tylerwmarrs) [inline matrixprofile/matrixProfile.py]: All of these checks will fail in the case of tsB being None. You need to check for a self join first and copy tsA to tsB.

```python
if not is_array_like(tsB):
    tsB = tsA
```

File: matrixprofile/matrixProfile.py [hunk 4]
```diff
@@ -122,6 +179,9 @@ def _matrixProfile_sampling(tsA,m,orderClass,distanceProfileFunction,tsB=None,sa
     order = orderClass(len(tsA)-m+1, random_state=random_state)
     mp, mpIndex = _self_join_or_not_preprocess(tsA, tsB, m)
```
Comment (tylerwmarrs) [inline matrixprofile/matrixProfile.py]: All of these checks will fail in the case of tsB being None. You need to check for a self join first and copy tsA to tsB.

```python
if not is_array_like(tsB):
    tsB = tsA
```

File: matrixprofile/matrixProfile.py [hunk 5]
```diff
@@ -151,6 +211,9 @@ def _matrixProfile_stomp(tsA,m,orderClass,distanceProfileFunction,tsB=None):
     order = orderClass(len(tsA)-m+1)
     mp, mpIndex = _self_join_or_not_preprocess(tsA, tsB, m)
```
Comment (tylerwmarrs) [inline matrixprofile/matrixProfile.py]: All of these checks will fail in the case of tsB being None. You need to check for a self join first and copy tsA to tsB.

```python
if not is_array_like(tsB):
    tsB = tsA
```

**Pull request merged**